sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/fl_scatter_gather.py

```diff
@@ -0,0 +1,705 @@
+                if k not in agg_outputs:
+                    validation_result.append_error(yaml_path="aggregation_to_silo_argument_map", message=f"aggregation_to_silo_argument_map key {k} is not a known output of the aggregation component.")
+                if v not in silo_inputs:
+                    validation_result.append_error(yaml_path="aggregation_to_silo_argument_map", message=f"aggregation_to_silo_argument_map value {v} is not a known input of the silo component.")
+        # and check the other mapping
+        if silo_to_aggregation_argument_map is None:
+            validation_result.append_error(yaml_path="silo_to_aggregation_argument_map", message=f"silo_to_aggregation_argument_map should never be None. Input an empty dictionary instead.")
+        elif agg_inputs is not None and silo_outputs is not None:
+            for k, v in silo_to_aggregation_argument_map.items():
+                if k not in silo_outputs:
+                    validation_result.append_error(yaml_path="silo_to_aggregation_argument_map", message=f"silo_to_aggregation_argument_map key {k} is not a known output of the silo component.")
+                if v not in agg_inputs:
+                    validation_result.append_error(yaml_path="silo_to_aggregation_argument_map", message=f"silo_to_aggregation_argument_map value {v} is not a known input of the aggregation component.")
+
+        # Check that inputs created by the node itself are valid
+        # inputs for the correct components
+        if pass_iteration_to_components:
+            if agg_inputs is not None and FL_ITERATION_INPUT not in agg_inputs:
+                validation_result.append_error(yaml_path="pass_iteration_to_components", message=f"pass_iteration_to_components is set to True, " + 
+                    f"but the aggregation component does not contain the required input '{FL_ITERATION_INPUT}'.")
+            if silo_inputs is not None and FL_ITERATION_INPUT not in silo_inputs:
+                validation_result.append_error(yaml_path="pass_iteration_to_components", message=f"pass_iteration_to_components is set to True, " + 
+                    f"but the silo component does not contain the required input '{FL_ITERATION_INPUT}'.")
+
+        if pass_index_to_silo_components:
+            if silo_inputs is not None and FL_SILO_INDEX_INPUT not in silo_inputs:
+                validation_result.append_error(yaml_path="pass_index_to_silo_components", message=f"pass_index_to_silo_components is set to True, " + 
+                    f"but the silo component does not contain the required input '{FL_SILO_INDEX_INPUT}'.")
+
+        if max_iterations < 1:
+            validation_result.append_error(yaml_path="max_iterations", message=f"max_iterations must be a positive value, not '{max_iterations}'.")
+
+
+        return validation_result.try_raise(
+            cls._get_validation_error_target(),
+            raise_error=raise_error,
+        )
```

Comment:
I'm ignorant to the actual requirements for this, but the length of this function makes me wonder if there's some way to leverage schema validation for this instead of doing it all from scratch.

Reply:
This object is never loaded from YAML, so doesn't that mean that schema validation isn't applicable here?